orchestration/dap_orchestration/solids.py

```diff
@@ -241,8 +242,57 @@ def write_outfiles(context: AbstractComputeExecutionContext, fan_in_results: lis
     NOTE: The fan_in_results param allows to introduce a fan-in dependency from upstream transformation
     solids, but is ignored by this solid.
     """
+    tsv_directory = context.solid_config["output_dir"]
     context.resources.outfiles_writer.run(
-        context.solid_config["output_dir"],
+        tsv_directory,
         context.resources.refresh_directory,
         fan_in_results
     )
+
+    return FanInResultsWithTsvDir(fan_in_results, tsv_directory)
+
+
+@solid(
+    required_resource_keys={"refresh_directory", "gcs"},
+    config_schema={
+        "destination_gcs_path": String,
+    }
+)
+def copy_outfiles_to_terra(
+    context: AbstractComputeExecutionContext,
+    surveyTypesWithTsvDir: FanInResultsWithTsvDir
+) -> None:
+    """
+    This solid will copy the tsvs created from write_outfiles to the provided GCS bucket. The script
+    will check that the file exists before uploading it and will error if it does not exist.
+
+    NOTE: The fan_in_results param allows to introduce a fan-in dependency from the upstream write_outfiles
+    solid and is used to iterate through TSV uploads for the surveys being refreshed.
+    # todo: handle having to upload with a different name as well? - sample.tsv + sample_MMDDYYYY.tsv
+    # todo: add firecloud flag to tsv script for sample outfile generation
+        # alternately we can hardcode entity:sample_id into the tsv script for the sample table
+        # the same could be done to handle the duplicate renamed file
+    """
+    for survey in surveyTypesWithTsvDir.fan_in_results:
+        storage_client = context.resources.gcs
+        tsv_dir = f"{surveyTypesWithTsvDir.tsv_dir}/tsv_output"
+        upload_dir = context.solid_config["destination_gcs_path"]
+
+        tsvBucketWithPrefix = parse_gs_path(tsv_dir)
+        destBucketWithPrefix = parse_gs_path(upload_dir)
+
+        outfilePrefix = f"{tsvBucketWithPrefix.prefix}/{survey}.tsv"
+        destOutfilePrefix = f"{destBucketWithPrefix.prefix}/{survey}.tsv"
+
+        source_bucket = storage_client.get_bucket(tsvBucketWithPrefix.bucket)
+        dest_bucket = storage_client.get_bucket(destBucketWithPrefix.bucket)
+
+        blob = source_bucket.get_blob(outfilePrefix)
+        if not blob.size:
+            raise Failure(f"Error; No {survey} files found in {tsv_dir}")
```

Comment:
Very small ⛏️ Add a line break for readability
```suggestion
            raise Failure(f"Error; No {survey} files found in {tsv_dir}")
            
```